add initial documentation for message system and creator site (#4735)
microsoft/fast: packages/tooling/fast-tooling/src/message-system/message-system.spec.md

# Message System

## Introduction

> The Message System controls the navigation, data, data schemas (JSON schemas) and history of messages it recieves. It can also be used to pass custom messages between services that are registered to it.

Reviewer: add a comma before 'and' and update spelling on "recieves".

> The Message System controls the navigation, data, data schemas (JSON schemas) and history of messages it recieves. It can also be used to pass custom messages between services that are registered to it.

Author: ```suggestion
The Message System controls the navigation, data, data schemas (JSON schemas), and history of messages it receives. It can also be used to pass custom messages between services that are registered to it.
```

                    schema: {
                        type: "object",
                        properties: {
                            disabled: {
                                type: "boolean"
                            }
                        }
                    },
                    disabled: false,
                    data: {
                        disabled: true
                    },
                    text: "Untitled",
                    type: DataType.object,
                    items: [
                        "disabled"
                    ],
                },
                "disabled": {
                    self: "disabled",
                    parent: "",
                    relativeDataLocation: "disabled",
                    schemaLocation: "properties.disabled",
                    schema: {
                        type: "boolean"
                    },
                    disabled: false,
                    data: true,
                    text: "Untitled",
                    type: DataType.boolean,
                    items: [],
                }
            },
            ""
        ]
    },
    "button"
]
```

> You can assume that to update a navigation you can use the dicionary ID and the navigation config ID, which for the above example if you wanted to navigate to the root of the button would be:

Reviewer: What about saying "To update navigation, use the dictionary ID and the navigation config ID.  The above example demonstrates navigation to the root of the button."

> You can assume that to update a navigation you can use the dicionary ID and the navigation config ID, which for the above example if you wanted to navigate to the root of the button would be:

Author: ```suggestion
To update navigation, use the dictionary ID and the navigation config ID. Using the data provided above as an example, to navigate to the root of the button, send the following message to the message system:
```

```js
{
    type: MessageSystemType.navigation;
    action: MessageSystemNavigationTypeAction.update;
    activeDictionaryId: "button";
    activeNavigationConfigId: "";
}
```

> And if you wanted to navigate to the `disabled` property, you would do:

Reviewer: "To navigate to the `disabled` property:"

> And if you wanted to navigate to the `disabled` property, you would do:

Author: ```suggestion
To navigate to the `disabled` property:
```
